### ISSUE TITLE
[FEA] Semantic versioning

### ISSUE THREAD

**Issue by sleeepyjack** [NVIDIA/cuCollections#190]
How do we want to employ semantic versioning for this library?

Thrust for instance updates the release version by hand by setting new values in [thrust/version.h](https://github.com/NVIDIA/thrust/blob/main/thrust/version.h).

This is required if we opt for versioned inline namespaces to solve #189.

**Comment by PointKernel** [comment]
cuCo doesn't have a release yet thus I'm not sure how to incorporate the versioned inline namespace in our case. e.g. how to name the version.

**Comment by sleeepyjack** [comment]
Yeah, that's true. Maybe we could push a development tag `0.1.0` to `dev` and use that as an intermediate?

**Comment by jrhemstad** [comment]
Inline namespaces won't solve https://github.com/NVIDIA/cuCollections/issues/189 in the short term because we aren't going to bump the version with every PR.

The best short term solution for https://github.com/NVIDIA/cuCollections/issues/189 is the custom namespace prefix.

**Comment by sleeepyjack** [comment]
Ok, then we can put this issue in the backlog for now.
